src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+            env(batch::outer(alice, seq, batchFee, tfIndependent),
+                batch::inner(pay(alice, bob, XRP(1)), seq + 1),
+                batch::inner(pay(alice, bob, XRP(2)), seq + 2),
+                batch::inner(pay(alice, bob, XRP(999)), seq + 3),
+                batch::inner(pay(alice, bob, XRP(3)), seq + 4),
+                ter(tesSUCCESS));
+            auto const txIDs = env.tx()->getBatchTransactionIDs();
+            TxID const parentBatchId = env.tx()->getTransactionID();
+            std::vector<TestBatchData> testCases = {
+                {"tesSUCCESS", to_string(txIDs[0])},
+                {"tesSUCCESS", to_string(txIDs[1])},
+                {"tecUNFUNDED_PAYMENT", to_string(txIDs[2])},
+                {"tesSUCCESS", to_string(txIDs[3])},
+            };
+            env.close();
+            validateBatch(env, parentBatchId, testCases);
+
+            // Alice consumes sequences (# of txns)
+            BEAST_EXPECT(env.seq(alice) == seq + 5);
+
+            // Alice pays XRP & Fee; Bob receives XRP
+            BEAST_EXPECT(env.balance(alice) == preAlice - XRP(6) - batchFee);
+            BEAST_EXPECT(env.balance(bob) == preBob + XRP(6));
+        }
+
+        // non tec error
+        {
+            auto const preAlice = env.balance(alice);
+            auto const preBob = env.balance(bob);
+
+            auto const batchFee = batch::calcBatchFee(env, 0, 4);
+            auto const seq = env.seq(alice);
+            auto tx3 = trust(alice, USD(1000), tfSetfAuth);
+            tx3[jss::Fee] = "0";
+            tx3[jss::Sequence] = seq + 3;
+            tx3[jss::SigningPubKey] = "";
+            tx3[jss::Flags] = tx3[jss::Flags].asUInt() | tfInnerBatchTxn;
+            env(batch::outer(alice, seq, batchFee, tfIndependent),
+                batch::inner(pay(alice, bob, XRP(1)), seq + 1),
+                batch::inner(pay(alice, bob, XRP(2)), seq + 2),
+                batch::inner_nofill(tx3),
+                // duplicate sequence
+                batch::inner(pay(alice, bob, XRP(3)), seq + 3),
+                ter(tesSUCCESS));
+            auto const txIDs = env.tx()->getBatchTransactionIDs();
+            TxID const parentBatchId = env.tx()->getTransactionID();
+            std::vector<TestBatchData> testCases = {
+                {"tesSUCCESS", to_string(txIDs[0])},
+                {"tesSUCCESS", to_string(txIDs[1])},
+                // tx #3 fails with tefNO_AUTH_REQUIRED
+                {"tesSUCCESS", to_string(txIDs[3])},
+            };
+            env.close();
+            validateBatch(env, parentBatchId, testCases);
+
+            // Alice consumes sequences (# of txns)
+            BEAST_EXPECT(env.seq(alice) == seq + 4);
+
+            // Alice pays XRP & Fee; Bob receives XRP
+            BEAST_EXPECT(env.balance(alice) == preAlice - batchFee - XRP(6));
+            BEAST_EXPECT(env.balance(bob) == preBob + XRP(6));
+        }
+    }
+
+    void
+    testBatchType(FeatureBitset features)
```

Comment:
How does this test differ from the ones above? Just an older version? Can we perhaps move these tests into those sections if that's the only difference?

Reply:
I can try to refactor them, they do check an interesting fill or kill use case that isnt checked elsewhere, but most of it seems to be a duplicate

Reply:
I'm fine if you just move the tests and keep the tests the same.